mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -737,11 +741,74 @@ def get_avg_latency(
             df.dropna(inplace=True)
         return df
 
+    async def add_basic_metrics(
+        self,
+        model_endpoint_objects: list[mlrun.common.schemas.ModelEndpoint],
+        project: str,
+        run_in_threadpool: Callable,
+    ) -> list[mlrun.common.schemas.ModelEndpoint]:
+        """
+        Add basic metrics to the model endpoint object.
+
+        :param model_endpoint_objects: A list of `ModelEndpoint` objects that will
+                                        be filled with the relevant basic metrics.
+        :param project:                The name of the project.
+        :param run_in_threadpool:      A function that runs another function in a thread pool.
+
+        :return: A list of `ModelEndpointMonitoringMetric` objects.
+        """
+
+        uids = [mep.metadata.uid for mep in model_endpoint_objects]
+        coroutines = [
+            run_in_threadpool(self.get_error_count, endpoint_ids=uids),
+            run_in_threadpool(self.get_last_request, endpoint_ids=uids),
+            run_in_threadpool(self.get_avg_latency, endpoint_ids=uids),
+            run_in_threadpool(self.get_drift_status, endpoint_ids=uids),
+        ]
+
+        (
+            error_count_df,
+            last_request_df,
+            avg_latency_df,
+            drift_status_df,
+        ) = await asyncio.gather(*coroutines)
```

Comment:
Can we put this in the base? to avoid code duplication.

Reply:
I'm going to leave the TDEngine issue for another PR since it's not a regression here, but trying to share this bit of code turns out too ugly for too little benefit, so probably not worth the trouble (plus, TDEngine code might change soon anyway).